src/mesh/column_mesh.jl

```diff
@@ -0,0 +1,242 @@
+#####
+##### Column mesh
+#####
+
+export n_cells, n_faces
+export ColumnMesh, coords
+export AbstractDataLocation, CellCent, CellFace
+export All, Ghost, Boundary, Interior
+export VerticalExtrema, VerticalMin, VerticalMax
+export n_hat, binary
+
+"""
+    AbstractDataLocation
+
+Subtypes are used for dispatching
+on the data location
+"""
+abstract type AbstractDataLocation end
+
+""" cell center location """
+struct CellCent <: AbstractDataLocation end
+
+""" cell face location """
+struct CellFace <: AbstractDataLocation end
+
+struct CollocatedCoordinates{DL <: AbstractDataLocation, A}
+    h::A
+    Δh::A
+end
+
+Base.length(c::CollocatedCoordinates) = length(c.h)
+
+function Base.show(io::IO, c::CollocatedCoordinates)
+    print(io, "\n")
+    println(io, "  size(h) = ", length(c))
+    println(io, "  h       = ", c.h)
+    println(io, "  Δh      = ", c.Δh)
+end
+
+"""
+    LocalStencil
+
+A local stencil, used for interpolations or
+finite-differences. This is effectively a
+single row of the global operator.
+"""
+struct LocalStencil{T}
```

Comment:
I think a stencil should encode in the type space:
- whether it is _from_ a centered or face field
- whether it is _to_ a centered or face field
- width and offset: this could be encoded as a "hi, lo" so that we can easily determine the required ghost cells